html/app.js

```diff
@@ -1048,14 +1062,7 @@ async function handle_message(event) {
     return;
   }
 
-  if (msg.type == "exception") {
-    lib.log(LOG_ERROR, "Exception thrown in audioworklet:", msg.exception);
-    if (app_state == APP_RUNNING) {
-      await reload_settings();
```

Comment:
There are currently two kinds of errors: recoverable ones (involve calls to `reload_settings()`) and unrecoverable ones (involve calls to `stop()`). It looks to me like you're collapsing both cases into unrecoverable?

Reply:
I'm treating all exceptions, except those thrown by fallible browser APIs and caught immediately at the call site, as unrecoverable. From conversations I had earlier, it didn't sound like we understood the state of all the possible invariants well enough to be able to safely use exceptions for flow control, or to create failure-isolation boundaries finer-grained than "reload the page". So for now, if users find exceptions, we should treat them as bugs and fix them (possibly by wrapping a browser API call in a `try` or `.catch` at the call site and then using a non-exception-based code path to handle that case). At some point it might make sense to try to set up failure-isolation boundaries so that the app can recover smartly from unanticipated exceptions, but right now that's likely to make its behavior harder to predict, which is the opposite of what we need.